### PR TITLE
entangled:  README - add information table + some desc

### LINKED ISSUE #269: No idea what this repo does.
### Issue
Looking at this repo and it's readme, tagline or tags I have no idea of the following:
1. What this repo is (apart from being a monorepo)
2. What the repo contains
3. What each folder does 
a. how to build each folder (except ccurl)
b. how the folder relates to each other folder 
c. folder's state of progress
d. folder's versioning

### Suggestions
1. `enTangle'd is an amalgamation of all things Tangle` is less than descriptive
  a. Suggest change to: `This is a monorepo containing all you need get IOTA operating in C`
  b. Meta overview of what this repo is, what is contains and its aims.
2. List what the repo contains
  a. Better descriptive tags for search-ability: `c` `c++` `c library`
  b. Create a `monoreadme`™ about the `monorepo` 
    - H3 Title for each folder
    - Two sentence paragraph describing the function of the folder
    - State of development (Active, abandoned, deprecated)
3. Readme in each folder containing:
  a. General description of the folder's content.  Copy pasta of the `monoreadme`™ description is fine if descriptive enough
  b. How to build each folder. If no build needed then list that.
  c. List how this folder depends on another folder in the repo
  d. The date of the latest commit and the version number (if one) of that commit. Preferably directly below the title and description. 
  e. If the folder is actively being developed or if its deprecated. 

I'm happy to assist on formatting

### PR DESCRIPTION
**Pull request by tsvisabo**
Fixes #269

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,20 @@
-# IOTA Entangled Monorepo
+## IOTA Entangled Monorepo
 [![Build status](https://badge.buildkite.com/42ef0512276a270bd34cb0010fd641558b344dab4aa2225fba.svg)](https://buildkite.com/iota-foundation/entangled)
 
+This is a monorepo containing all you need get [[IOTA]](https://iota.org/) operating in C/C++
```
Comment (thibault-martinez) [inline README.md]: Typo:  `all you need TO get`

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (tsvisabo) [reply README.md]: Worst kind of dyslexia :)

File: common/helpers/sign.c [hunk 2]
```diff
@@ -44,7 +45,7 @@ IOTA_EXPORT char* iota_sign_address_gen(char const* const seed,
   iss_kerl_subseed(subseed, subseed, index, &kerl);
 
   iss_kerl_key(subseed, key, key_length, &kerl);
-  memset(subseed, 0, HASH_LENGTH * sizeof(trit_t));
+  memset_s(subseed,HASH_LENGTH, 0, HASH_LENGTH * sizeof(trit_t));
```
Comment (oopsmonk) [inline common/helpers/sign.c]: This is no relation with this PR and no clang-format applied.

File: common/helpers/sign.c [hunk 3]
```diff
@@ -87,7 +88,7 @@ IOTA_EXPORT char* iota_sign_signature_gen(char const* const seed,
   trytes_to_trits((tryte_t*)bundle_hash, hash, HASH_LENGTH / RADIX);
   iss_kerl_subseed(subseed, subseed, index, &kerl);
   iss_kerl_key(subseed, key, key_length, &kerl);
-  memset(subseed, 0, HASH_LENGTH * sizeof(trit_t));
+  memset_s(subseed,HASH_LENGTH, 0, HASH_LENGTH * sizeof(trit_t));
```
Comment (oopsmonk) [inline common/helpers/sign.c]: This is no relation with this PR and no clang-format applied.

File: common/helpers/sign.c (hunk 2 above)
Comment (tsvisabo) [reply common/helpers/sign.c]: thanks, this is a mistake

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**